### PR TITLE
subcommunities: added base request and service.

### PR DESCRIPTION
**Pull request by alejandromumo**
closes https://github.com/inveniosoftware/invenio-communities/issues/1147

### CODE REVIEW

File: invenio_communities/communities/services/components.py [hunk 1]
```diff
@@ -128,6 +129,11 @@ def create(self, identity, data=None, record=None, **kwargs):
         )
 
         # Invalidate the membership cache
+        # The identity is updated so it can be used immediately after the creation of the community
+        # Otherwise, the identity only gets updated in the next request
+        identity.provides.add(
+            CommunityRoleNeed(str(record.id), current_roles.owner_role.name)
```
Comment (alejandromumo) [inline invenio_communities/communities/services/components.py:135]: For reviewers: this is a significant change, please have a look at it

File: invenio_communities/views/__init__.py [hunk 2]
```diff
@@ -13,12 +13,13 @@
...
+    create_subcommunities_api_blueprint,
 )
 from .ui import create_ui_blueprint
 
 __all__ = (
     "blueprint",
     "create_communities_api_blueprint",
     "create_members_api_bp_from_app",
-    "create_ui_blueprint",
+    "create_subcommunities_api_blueprint" "create_ui_blueprint",
```
Comment (Samk13) [inline invenio_communities/views/__init__.py:24]: Missing comma?

File: tests/communities/test_subcommunities.py [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+        "is_expired": False,
+        "is_open": False,
+        "links": {
+            "actions": {},
+            "comments": f"https://127.0.0.1:5000/api/requests/{request_id}/comments",
+            "self": f"https://127.0.0.1:5000/api/requests/{request_id}",
+            "self_html": f"https://127.0.0.1:5000/requests/{request_id}",
+            "timeline": f"https://127.0.0.1:5000/api/requests/{request_id}/timeline",
+        },
+    }
```
Comment (slint) [inline tests/communities/test_subcommunities.py:269]: minor: check that child community has `parent_id` set.

File: invenio_communities/views/__init__.py (hunk 2 above)
Comment (slint) [reply invenio_communities/views/__init__.py:24]: Thanks, nice catch! 🦅

**Pull request merged**